climakitae/explore/amy.py

```diff
@@ -111,6 +114,124 @@ def _convert_stations_to_lat_lon(
 
 
+def _get_clean_standardyr_filename(
+    var_id: str, q: float, location: str, gwl: float, no_delta: bool
+) -> str:
+    """
+    Standardizes filename export for standard year files
+
+    Parameters
+    ----------
+    var_id : str
+        Name of variable used in profile
+    q : float
+        Percentile used in profile
+    location: str
+        String describing profile location
+    gwl : float
+        Single gwl for csv file name
+    no_delta : bool
+        no_delta value used to generate profile
+
+    Returns
+    -------
+    str
+        A cleaned up file name
+    """
+
+    # clean arguments for filenaming
+    clean_loc_name = location.replace(" ", "_").replace("(", "").replace(")", "")
+    clean_q_name = f"{q:.2f}".split(".")[1]
+    clean_var_name = var_id
+    clean_gwl_name = match_str_to_wl(gwl)
+    if no_delta:
+        delta_str = ""
+    else:
+        delta_str = "_delta_from_historical"
+
+    filename = (
+        f"stdyr_{clean_var_name}_{clean_q_name}ptile_{clean_loc_name}_{clean_gwl_name}{delta_str}.csv"
+    ).lower()
+    return filename
+
+
+def export_profile_to_csv(
+    profile: pd.DataFrame,
+    variable: str,
+    q: float,
+    global_warming_levels: list[float],
+    station_name: str = UNSET,
+    cached_area: str = UNSET,
+    latitude: float = UNSET,
+    longitude: float = UNSET,
+    no_delta: bool = False,
+):
+    """Export profile to csv file with a descriptive file name.
+
+    Each warming level is saved in a separate file.
+
+    Parameters
+    ----------
+    profile : pd.DataFrame
+        Standard year profile with MultiIndex columns
+    variable : str
+        Name of variable used in profile
+    q : float
+        Percentile used in profile
+    global_warming_levels : list[float]
+        List of global warming levels in profile
+    station_name : str, optional
+        Name of station used in profile
+    cached_areas : str, optional
+        Name of cached area used in profile
+    latitude : float
+        Latitude coordinate from profile location
+    longitude : float
+        Longitude coordinate from profile location
+    no_delta : bool, optional
+        True if no_delta=True when generating profile
+
+    """
+
+    # Get variable id string to use in file name
+    variable_descriptions = read_csv_file(VARIABLE_DESCRIPTIONS_CSV_PATH)
+    var_id = variable_descriptions[
+        (variable_descriptions["display_name"] == variable)
+        & (variable_descriptions["timescale"] == "hourly")
+    ]["variable_id"].item()
+
+    # Get location string
+    if station_name is not UNSET:
+        location_str = station_name
+    elif (latitude is not UNSET) and (longitude is not UNSET):
+        location_str = f"{latitude}N_{longitude}E"
+    elif cached_area is not UNSET:
+        location_str = cached_area
+    else:
+        raise TypeError(
+            "Location must be provided as either `station_name` or `cached_area` or `latitude` plus `longitude`."
+        )
+
```

Comment:
V2 edge case to handle: a user passes in a lat-lon location of a particular asset / plant, but they want the name to be "custom". E.g., "asset 369" is at lat/lon 40.404N -122.222W so they pass those coords as the lat/lon but they want the filename to be "_asset369_" etc.

Reply:
Ok I've reworked this to be a bit more like TMY. `station_name` can be used for a custom name along with lat/lon as long as it's not a HadISD station. `cached_area` is an additional variable that can be used only by itself.